spec/rspec/core/example_status_persister_spec.rb

```diff
@@ -58,6 +58,37 @@ def new_example(id, metadata = {})
         )
       end
 
+      it 'prevents simultaneous access to the file' do
+        # This tests whether a certain race condition is prevented:
+        #  - read 1
+        #  - read 2
+        #  - write 1
+        #  - write 2 - write 1 is lost
+        ex_1 = new_example("#{existing_spec_file}[1:1]", :status => :passed)
+        ex_2 = new_example("spec_1.rb[1:1]", :status => :failed)
+
+        persister_1 = ExampleStatusPersister.new([ex_1], file.path)
+        persister_2 = ExampleStatusPersister.new([ex_2], file.path)
+        persister_1_dumped_statuses = persister_1.method(:dumped_statuses)
+        persister_2_thread = nil
+
+        # dumped status is called after the file is locked but before
+        # the output is written
+        allow(persister_1).to receive(:dumped_statuses) do
```

Comment:
This can use `receive(:dumped_statuses).and_wrap_original` rather than capturing the method above.

https://relishapp.com/rspec/rspec-mocks/v/3-3/docs/configuring-responses/wrapping-the-original-implementation